src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/IOpenIdConnectEvents.cs

```diff
@@ -38,17 +28,22 @@ public interface IOpenIdConnectEvents : IRemoteAuthenticationEvents
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        Task RedirectToAuthenticationEndpoint(RedirectContext context);
+        Task RedirectToIdentityProvider(RedirectContext context);
 
         /// <summary>
         /// Invoked before redirecting to the identity provider to sign out.
         /// </summary>
-        Task RedirectToEndSessionEndpoint(RedirectContext context);
+        Task RedirectToIdentityProviderForSignOut(RedirectContext context);
 
         /// <summary>
         /// Invoked after "authorization code" is redeemed for tokens at the token endpoint.
         /// </summary>
-        Task TokenResponseReceived(TokenResponseReceivedContext context);
+        Task TokenResponseMessageReceived(TokenResponseMessageReceivedContext context);
```

Comment:
@HaoK are you kidding or are you serious? If the latter, could you provide a reference to the rule? :)
@PinpointTownes that's for being coherent with messagereceived - that should help the developer to understand what kind (as in class) of event they are receiving.